codalab/worker/main.py

```diff
@@ -298,8 +299,15 @@ def parse_gpuset_args(arg):
         return set()
 
     try:
-        all_gpus = docker_utils.get_nvidia_devices()  # Dict[GPU index: GPU UUID]
-    except docker_utils.DockerException:
+        nvidia_command = ['nvidia-smi', '--query-gpu=index,uuid', '--format=csv,noheader']
```

Comment:
Should add a comment that says we shouldn't trust docker (e.g., in slurm).  Is this a known docker bug?

Reply:
Yeah, I'll add a comment. Unclear if this is a bug or a feature, though—it's the same deal with cpuset, right? Docker just in general runs independent of most (any?) host-level restrictions.